build/android/app/src/main/java/net/minetest/minetest/CopyZipTask.java

```diff
@@ -0,0 +1,91 @@
+/*
+Minetest
+Copyright (C) 2014-2019 MoNTE48, Maksim Gamarnik <MoNTE48@mail.ua>
+Copyright (C) 2014-2019 ubulem,  Bektur Mambetov <berkut87@gmail.com>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+package net.minetest.minetest;
+
+import android.annotation.SuppressLint;
+import android.content.Intent;
+import android.os.AsyncTask;
+import android.util.Log;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+class CopyZipTask extends AsyncTask<String, Void, String> {
+	@SuppressLint("StaticFieldLeak")
+	private final AppCompatActivity activity;
+	private boolean isCancel = false;
+
+	CopyZipTask(AppCompatActivity activity) {
+		this.activity = activity;
+	}
+
+	protected String doInBackground(String... params) {
+		while (!isCancel)
+			copyAssets(params);
+		return params[0];
+	}
+
+	@Override
+	protected void onPostExecute(String result) {
+		startUnzipService(result);
+	}
+
+	private void copyAsset(String zipName) throws IOException {
```

Comment:
1) Your code is broken. 
2) We already have throws an IOException in the original function.
This code is used in MultiCraft, and there has never been a bug report from my analytics.

Reply:
throwing an exception would result in the streams never being closed

Reply:
I didn't test the code because Android Studio was still installing. The fix is replacing `,` with `;` like so:

```java
private void copyAsset(String zipName) throws IOException {
	String filename = zipName.substring(zipName.lastIndexOf("/") + 1);
	try (InputStream in = activity.getAssets().open(filename);
			OutputStream out = new FileOutputStream(zipName)) {
		copyFile(in, out);
	}
}
```